portfolio/src/main/webapp/script.js

```diff
@@ -40,27 +40,37 @@ function highlightTab() {
 }
 
 function fetchComments() {
-  (fetch('/comments').then(response => response.json()).then((comments) => {
+  // get num of comments to query
+  const selectBar = document.getElementById("numComments");
+  const maxComments = selectBar.options[selectBar.selectedIndex].value;
+
+  (fetch('/comments?numComments=' + maxComments).then(response => response.json()).then((comments) => {
+    // clear data-container
+    const container = document.getElementById("data-container")
+    while (container.firstChild) {
```

Comment:
I was curious on what the best way to empty an element is. I found this:
https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript

Looks there is no really best way and your way is one of the good options.

Reply:
Personally - I'd prefer `container.innerHTML = ''` since it's the simplest/easiest to read. Optimizing for performance here isn't really needed with so few elements so readability is more important.

Just my take :)